### PR TITLE
fix sub-domain cookie

### LINKED ISSUE #71: This attempt to set a cookie via the Set-Cookie header was blocked....
I am running mercure from a different subdomain
mercure.product.ota.internal.company.cloud

the app is running from
app.product.ota.internal.company.cloud

Whenever I want to use a discover as shown in the guide I get the following error within Edge (Blink) 95.0

image

I seen #54 and updated to 0.6.0, so most recent version

mercureAuthorization=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MzU4NzMxNjcuNTgzMTY3MSwibWVyY3VyZSI6eyJwdWJsaXNoIjpbXSwic3Vic2NyaWJlIjpbIi9wcm9kdWN0aW9uL3NjZW5hcmlvLzIiXX19.8rdaR3akCcobIpCb9aBE7DptsuMvC_hsrzvEEePW5A4; path=/.well-known/mercure; domain=mercure.product.ota.internal.company.cloud; secure; httponly; samesite=strict

Am I reading something wrong or is this a browser issue?

### PR DESCRIPTION
**Pull request by azjezz**
closes #71

### CODE REVIEW

File: tests/AuthorizationTest.php [hunk 1]
```diff
@@ -123,9 +123,11 @@ function (Update $u): string { return 'dummy'; },
...
     public function provideApplicableCookieDomains(): iterable
     {
-        yield ['demo.example.com', 'https://demo.example.com', 'https://example.com'];
-        yield ['mercure.example.com', 'https://mercure.example.com', 'https://app.example.com'];
-        yield ['example.com', 'https://example.com/.well-known/mercure', 'https://app.example.com'];
+        yield ['.example.com', 'https://foo.bar.baz.example.com', 'https://foo.bar.baz.qux.example.com'];
+        yield ['.foo.bar.baz.example.com', 'https://mercure.foo.bar.baz.example.com', 'https://app.foo.bar.baz.example.com'];
+        yield ['example.com', 'https://demo.example.com', 'https://example.com'];
+        yield ['.example.com', 'https://mercure.example.com', 'https://app.example.com'];
+        yield ['.example.com', 'https://example.com/.well-known/mercure', 'https://app.example.com'];
```
Comment (azjezz) [inline tests/AuthorizationTest.php:130]: This one I'm not sure about.

in [RFC2109](www.ietf.org/rfc/rfc2109.txt) it says:

```text
   * A is a FQDN string and has the form NB, where N is a non-empty name
     string, B has the form .B', and B' is a FQDN string.  (So, x.y.com
     domain-matches .y.com but not y.com.)
```

cc @dunglas

Comment (azjezz) [reply tests/AuthorizationTest.php:130]: and we can't set the cookie for example.com from app.example.com 😶

Comment (dunglas) [reply tests/AuthorizationTest.php:130]: This RFC is outdated, see https://stackoverflow.com/a/23086139/1352334 and https://stackoverflow.com/a/23086139/1352334 for the current rules.

So this works.

**Comment by tchapi** [comment]
👋🏼 

Is there anything we can do to help get this PR merged @dunglas ? I'm available if needed.

Thanks a lot !

**Comment by dunglas** [comment]
Thanks @azjezz!

**Pull request merged**